src/docfx/config/Config.cs

```diff
@@ -193,68 +197,89 @@ private static void ValidateContributorConfig(ContributionConfig config, string
             }
         }
 
-        private static JObject LoadOriginalConfigObject(string configPath, List<string> parents, bool expand)
+        private static JObject LoadConfigObject(string docsetPath, string filePath, bool extend, RestoreMap restoreMap)
         {
-            // TODO: support URL
-            var (errors, config) = YamlUtility.Deserialize<JObject>(File.ReadAllText(configPath));
+            var (errors, config) = YamlUtility.Deserialize<JObject>(File.ReadAllText(filePath));
             if (errors.Any())
             {
                 throw errors[0].ToException();
             }
 
             if (config == null)
                 config = new JObject();
-            if (!expand || !config.TryGetValue(ConfigConstants.Extend, out var objExtend))
+
+            if (!extend)
                 return config;
 
-            if (parents.Contains(configPath))
```

Comment:
what is the output of:

docfx.yml:
  extends: docfx.yml

Reply:
just itself as we don't support deep level extends.

---
In reply to: [203657706](https://github.com/dotnet/docfx/pull/3101#discussion_r203657706) [](ancestors = 203657706)

Reply:
i can add a test for that

---
In reply to: [203660745](https://github.com/dotnet/docfx/pull/3101#discussion_r203660745) [](ancestors = 203660745,203657706)